consensus/src/experimental/buffer_manager.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use consensus_types::executed_block::ExecutedBlock;
+use diem_types::ledger_info::LedgerInfoWithSignatures;
+use futures::channel::{mpsc::UnboundedReceiver, oneshot};
+use std::{collections::LinkedList, sync::Arc};
+
+pub type ResetAck = ();
+pub fn reset_ack_new() -> ResetAck {}
+
+pub struct ResetEventType {
+    tx: oneshot::Sender<ResetAck>,
+    reconfig: bool,
+}
+
+pub struct BufferManagerType {
+    pub blocks: Vec<ExecutedBlock>,
+    pub finality_proof: LedgerInfoWithSignatures,
+}
+
+/// BufferManager handles a buffer of executed blocks and
+/// interacts with the execution phase, the signing phase, and
+/// the persisting phase.
+pub struct BufferManager {
```

Comment:
I'd call it StateManager to reflect it's managing operations around the account state.